### DISCUSSION TITLE
Using AWS Batch as an all-in-one ActiveJob solution?

### DISCUSSION THREAD

**Discussion by OmriSama** [aws/aws-sdk-ruby#2687] [Q&A]
Hi @mullermp and team! Great work on the SDK as always :)

I've been learning more recently about AWS Batch and it sounds like it could potentially be using as an all-in-one solution for ActiveJob. It doesn't only handle/manage the abstraction layer over the EC2 ASG job processes, but it also includes a Job Queue.

I know about using SQS as a queueing mechanism, but has anyone looked at Batch before? Is this possible?

**Comment by mullermp** [comment]
Thank you, and thanks for opening up a discussion. I have never used AWS Batch but on the surface I think it can certainly be used in an ActiveJob adapter.

We have a gem for Rails integrations (https://github.com/aws/aws-sdk-rails) that could have such an adapter. Would you mind opening up a feature request there? Also.. contributions are certainly welcome. aws-sdk-rails implements an SQS ActiveJob adapter that can be used as a starting point if you were taking this on yourself. (see: https://github.com/aws/aws-sdk-rails/blob/main/lib/active_job/queue_adapters/amazon_sqs_adapter.rb)
